lib/src/simple_op_heads_store.rs

```diff
@@ -236,3 +254,65 @@ impl OpHeadsStore for SimpleOpHeadsStore {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashSet;
+    use std::fs;
+    use std::path::Path;
+
+    use itertools::Itertools;
+
+    use super::InnerSimpleOpHeadsStore;
+    use crate::op_heads_store::OpHeadsStore;
+    use crate::op_store::OperationId;
+    use crate::simple_op_heads_store::SimpleOpHeadsStore;
+
+    fn read_dir(dir: &Path) -> Vec<String> {
+        fs::read_dir(dir)
+            .unwrap()
+            .map(|entry| entry.unwrap().file_name().to_str().unwrap().to_string())
+            .sorted()
+            .collect()
+    }
+
+    #[test]
+    fn test_simple_op_heads_store_migration() {
+        let test_dir = testutils::new_temp_dir();
+        let store_path = test_dir.path().join("op_heads");
+        fs::create_dir(&store_path).unwrap();
+
+        let op1 = OperationId::from_hex("012345");
+        let op2 = OperationId::from_hex("abcdef");
+        let mut ops = HashSet::new();
+        ops.insert(op1.clone());
+        ops.insert(op2.clone());
+
+        let old_store = InnerSimpleOpHeadsStore {
+            dir: store_path.clone(),
+        };
+        old_store.add_op_head(&op1);
+        old_store.add_op_head(&op2);
+
+        assert_eq!(vec!["012345", "abcdef"], read_dir(&store_path));
+        drop(old_store);
```

Comment:
Why bother to explicitly drop it (and `new_store` below)? There's no lock held, right? And it doesn't seem necessary in order to please the borrow checker.

Reply:
I'm not sure; done.